app/controllers/home/oauths_controller.rb

```diff
@@ -10,17 +8,17 @@ def oauth
   def callback
     provider = auth_params[:provider]
     if @user = login_from(provider)
-      redirect_to root_path, notice: (t 'log_in_is_successful_provider_notice',
+      redirect_to trainer_path, notice: (t 'log_in_is_successful_provider_notice',
                                         provider: provider.titleize)
     else
       begin
         @user = create_from(provider)
         reset_session
         auto_login(@user)
-        redirect_to root_path, notice: (t 'log_in_is_successful_provider_notice',
+        redirect_to trainer_path, notice: (t 'log_in_is_successful_provider_notice',
                                           provider: provider.titleize)
       rescue
-        redirect_to root_path, alert: (t 'log_out_failed_provider_alert',
+        redirect_to user_sessions_path, alert: (t 'log_out_failed_provider_alert',
```

Comment:
Line is too long. [82/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.